pkg/dataloader/variantsyncer/variantsyncer.go

```diff
@@ -0,0 +1,65 @@
+package variantsyncer
+
+import (
+	"context"
+	"reflect"
+
+	log "github.com/sirupsen/logrus"
+
+	bqcached "github.com/openshift/sippy/pkg/bigquery"
+	"github.com/openshift/sippy/pkg/db"
+	"github.com/openshift/sippy/pkg/db/models"
+	"github.com/openshift/sippy/pkg/testidentification"
+)
+
+type VariantSyncer struct {
+	dbc    *db.DB
+	mgr    testidentification.VariantManager
+	errors []error
+}
+
+func New(dbc *db.DB, bqc *bqcached.Client) (*VariantSyncer, error) {
+	mgr, err := testidentification.NewOpenshiftVariantManager(context.TODO(), bqc)
+	if err != nil {
+		return nil, err
+	}
+
+	return &VariantSyncer{
+		dbc: dbc,
+		mgr: mgr,
+	}, nil
+}
+
+func (vl *VariantSyncer) Name() string {
+	return "sync-variants"
+}
+
+func (vl *VariantSyncer) Errors() []error {
+	return vl.errors
+}
+
+func (vl *VariantSyncer) Load() {
+	allJobs := loadAllProwJobs(vl.dbc)
+	for _, j := range allJobs {
+		newVariants := vl.mgr.IdentifyVariants(j.Name, j.Release, models.ClusterData{})
+		if !reflect.DeepEqual(newVariants, []string(j.Variants)) {
+			j.Variants = newVariants
+		}
+		if res := vl.dbc.DB.WithContext(context.TODO()).Save(j); res.Error != nil {
```

Comment:
Should that jump up into the DeepEqual if? Can't recall if that's a no-op in gorm if nothing has changed.